lib/Sema/LookupVisibleDecls.cpp

```diff
@@ -801,7 +699,9 @@ class OverrideFilteringConsumer : public VisibleDeclConsumer {
 
     // If this kind of declaration doesn't participate in overriding, there's
     // no filtering to do here.
-    if (!isa<AbstractFunctionDecl>(VD) && !isa<AbstractStorageDecl>(VD)) {
+    if (!isa<AbstractFunctionDecl>(VD) &&
+        !isa<AbstractStorageDecl>(VD) &&
+        !isa<AssociatedTypeDecl>(VD)) {
```

Comment:
Technically, associated types do have "overrides" (when an associated type is restated in an inheriting protocol).

Reply:
Yeah, notice the 'if' check is an early return, so we now handle associated types whereas before we skipped them.

Previously the RestatedFilteringConsumer would catch restated associated types, but now it has to be handled in the OverrideFilteringConsumer.